examples/demo/Extras/Tree_editor_with_TreeNodeRenderer.py

```diff
@@ -0,0 +1,113 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2019, Enthought, Inc.
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+#
+# -----------------------------------------------------------------------------
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from random import choice
+
+import numpy as np
+
+from pyface.qt import QtCore, QtGui
+from traits.api import Array, HasTraits, Instance, List, Unicode
+
+from traitsui.api import TreeEditor, TreeNode, UItem, View
+from traitsui.tree_node_renderer import AbstractTreeNodeRenderer
+from traitsui.qt4.tree_editor import WordWrapRenderer
+
+
+class MyDataItem(HasTraits):
+    text = Unicode
+
+    data = Array
+
+    def _data_default(self):
+        return np.random.standard_normal((1000,)).cumsum()
+
+
+class MyData(HasTraits):
+    name = Unicode('Rooty McRootface')
+    items = List(Instance(MyDataItem))
```

Comment:
(suggestion) replace `items` with `elements`. To avoid naming confusion with `_items` postfix on traits notification expressions.